batch/batch/worker/worker.py

```diff
@@ -361,6 +365,18 @@ async def run(self, worker):
                 self.container = await docker_call_retry(MAX_DOCKER_OTHER_OPERATION_SECS, f'{self}')(
                     create_container, config, name=f'batch-{self.job.batch_id}-job-{self.job.job_id}-{self.name}')
 
+            c = await docker_call_retry(MAX_DOCKER_OTHER_OPERATION_SECS, f'{self}')(self.container.show)
+
+            merged_overlay_path = c['GraphDriver']['Data']['MergedDir']
+            assert merged_overlay_path.endswith('/merged')
+            self.overlay_path = merged_overlay_path[:-7].replace(LOCAL_SSD_MOUNT, '/host')
+            os.makedirs(f'{self.overlay_path}/', exist_ok=True)
+
+            await check_shell(f'''
+python3 -m batch.worker.flock /xfsquota/projects -c "echo \"{self.job.project_id}:{self.overlay_path}\" >> /xfsquota/projects"
```

Comment:
Hmm. This feels a bit weird. You you are shelling out from python to python. Is the issue that the different async threads might race to update the projects file? It seems more readable to do something like:

```python3
with Flock('/xfsquota/projects'):  # maybe need to also use a python lock to block other async threads
    with open('/xfsquota/projects', 'a') as f:
        f.write(f'{self.job.project_id}:{self.overlay_path}\n')
```

Reply:
I'm worried that the Flock command isn't async. I think I need an async implementation... sigh.